crates/fluvio-storage/src/replica.rs

```diff
@@ -97,6 +99,39 @@ impl ReplicaStorage for FileReplica {
         }
     }
 
+    /// reads log file of the current partition
+    /// return the size in byte
+    #[instrument(skip(self))]
+    async fn get_partition_size(&self) -> Result<u64, ErrorCode> {
+        let mut entries = read_dir(&self.option.base_dir).await.map_err(|err| {
+            error!(
+                "failed to open the log base dir \"{}\": {}",
+                self.option.base_dir.to_string_lossy(),
+                err
+            );
+            ErrorCode::StorageError
+        })?;
+
+        while let Ok(Some(entry)) = entries.try_next().await {
+            let os_file_name = entry.file_name();
+            if os_file_name.to_string_lossy().ends_with(".log") {
+                let metadata = entry.metadata().await.map_err(|err| {
+                    error!(
+                        "fetching metadata for file \"{}\" failed: {err}",
+                        os_file_name.to_string_lossy(),
+                    );
+                    ErrorCode::StorageError
+                })?;
+                return Ok(metadata.len());
```

Comment:
This is only returning the size of the first log file that it gets. We can have multiple segments in the topic, so I think that we should return the sum of the size of all of them. 

You can test this by using a topic with a low segment-size: `fluvio topic create topicname --segment-size 1024`, once the first log file reaches 1024, it creates a new .log file.

Not sure if we also should count the size of the replication.chk and .index files, what do you think @sehz ?

Reply:
replication and index are internal file so let's not expose them yet (and smaller too)